compiler/src/main/java/com/raizlabs/android/dbflow/processor/utils/WriterUtils.java

```diff
@@ -71,4 +74,29 @@ public static boolean writeBaseDefinition(BaseDefinition baseDefinition, Process
         return success;
     }
 
+    /**
+     * Checks if className is valid Java class name. It will check for Java keywords like abstract, assert, boolean etc.
+     * and it will check if className matches regex pattern [A-Za-z_$]+[a-zA-Z0-9_$]*
```

Comment:
#req I would say give an example of what it takes to pass the regex, and maybe some failure examples.